packages/listview/src/components/ColoredStatusColumn/index.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import useTheme from 'react-rainbow-components/libs/hooks/useTheme.js';
+import getContrastText from 'react-rainbow-components/styles/helpers/color/getContrastText.js';
+import { StyledContainer } from './styled';
+
+const ColoredStatusColumn = ({ value, colors, textTransform }) => {
+    const theme = useTheme();
+
+    const { success, warning, error } = theme.rainbow.palette;
+    const defaultColors = {
+        success: success.main,
+        warning: warning.main,
+        error: error.main,
+    };
+
+    const map = {
+        ...defaultColors,
+        ...colors,
+    };
+
+    let background;
+    let color;
+    if (map[value]) {
+        if (map[value].backgroundColor) {
+            background = map[value].backgroundColor;
+        } else {
+            background = map[value];
+        }
+        if (map[value].color) {
+            color = map[value].color;
+        } else {
+            color = getContrastText(background);
+        }
+    }
```

Comment:
move this logic into functions that you can test and it will simplify the complexity e.g.

```js
const backgroundColor = getBackgrondColor(value);
const color = getColor(value);

const getBackgroundColor = (value) => {
    if (typeof map[value] === 'string' && isValidColor(map[value])) {
        return map[value];
    } 
    If (map[value] && typeof map[value].backgroundColor === 'string' && isValidColor(map[value].background)) {
        return map[value].backgroundColor;
    }
    return defaultBackgroundColor; // white
}

const getColor(value) => {
    if (map[value] && typeof map[value].color === 'string' && isValidColor(map[value].color)) {
       return map[value].color;
    }
    const backgroundColor = getBackgroundColor(value);
    return getContrastText(backgroundColor);
}
```